include/alpaka/mem/alloc/AllocCpuBoostAligned.hpp

```diff
@@ -51,10 +53,24 @@ namespace alpaka
                         std::size_t const & sizeElems)
                     -> T *
                     {
+#if (defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && BOOST_LANG_CUDA) || (defined(ALPAKA_ACC_GPU_HIP_ENABLED) && BOOST_LANG_HIP)
+                        // For CUDA host memory must be aligned to 4 kib to pin is with `cudaHostRegister`,
+                        // this was described in older programming guides but was removed later.
+                        // From testing with PIConGPU and cuda-memcheck we found out that the alignment is still required.
+                        //
+                        // For HIP the required alignment is the size of a cache line.
+                        // https://rocm-developer-tools.github.io/HIP/group__Memory.html#gab8258f051e1a1f7385f794a15300e674
+                        // To avoid issues with HIP(cuda) the alignment will be set also for HIP(clang/hcc)
+                        // to 4kib.
+                        // @todo evaluate requirements when the HIP ecosystem is more stable
+                        constexpr size_t minAlignement = 4096;
```

Comment:
Could we change this to `std::max(TAlignment::value, 4096u)` and remove the std::max below so that we do not do `std::max(TAlignment::value, TAlignment::value)` for the other case? Compilers might probably optimize this but I think this would be clearer.

Reply:
I would like to keep it because this has already the structure to handle HIP and CUDA different and it is easy to read.
With C++14 `std::max` is constexpr and before it is one comparison before a very expensive malloc call.

Reply:
I like a separate `minAlignement` variable, since it has a clear meaning on its own and the limitations are defined in terms of this value. I would prefer additionally to make another variable, like `[constexpr in case it's allowed] alignment = std::max(TAlignment::value, minAlignement);` and use it later for the `aligned_alloc` call.

Reply:
You can keep it as is and still make it more readable and compile-time in C++11 by writing:
```c++
constexpr size_t minAlignement = ...;

constexpr size_t alignment = (TAlignment::value > minAlignement) ? TAlignment::value : minAlignement;

boost::alignment::aligned_alloc(alignment, ...);
```